### PR TITLE
Add watcher thread to report shutdown steps.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

- Will add a timeout wait for completions in a follow up PR.

##### Test Plan

- CI jobs.
- Build with/without DBENGINE and with/without a normal exit code. The agent should stop running in all cases.

##### Additional Information

The main drawback of this approach is that the number of steps is fixed and they have to be reported regardless of whether the actually occur in the code path when exiting.

However, this complication only occurs once when we decided how to shutdown the agent based on the `ret` code and whether `dbengine` is enabled/disabled.

### CODE REVIEW

**Comment by vkalintiris** [comment]
@stelfrag Marked as draft because I want your feedback on whether the high-level design is ok or not.

File: src/daemon/watcher.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    UNUSED(arg);
+
+    netdata_log_debug(D_SYSTEM, "Watcher thread started");
+
+    completion_wait_for(&shutdown_begin_completion);
+    usec_t shutdown_start_time = now_monotonic_usec();
+
+    // TODO:
+    //   - add a version of completion_wait_for with timeout
+    //   - check the step's duration and abort when the timeout has expired.
```
Comment (thiagoftsm) [inline src/daemon/watcher.c:35]: I suppose this `TODO` will come with another PR, please, can you confirm this?

**Review by thiagoftsm — APPROVED** [review]
My doubt is not a blocker, so I am approving the PR to avoid block next steps. 
PR is working as expected, LGTM!

File: src/daemon/watcher.c (hunk 1 above)
Comment (vkalintiris) [reply src/daemon/watcher.c:35]: Correct.

**Pull request merged**